File: test_flutter/lib/data/orders_repository.dart
import 'models/order_model.dart';

class OrdersRepository {
  Future<dynamic> fetchOrders() async {
    List<OrderModel> orders = [
      OrderModel(
          object: "dfdfd", guests: 3, table: '2', date: '', id: 2, items: []),
      OrderModel(
          object: "dfdfd", guests: 3, table: '2', date: '', id: 2, items: []),
      OrderModel(
          object: "sd", guests: 2, table: '6', date: '', id: 1, items: [])
    ];
    return orders;
  }
}

File: test_flutter/ios/AppDomain/OrdersRepository.swift
//
//  OrdersRepository.swift
//  AppDomain
//
//  Created by Ibrokhim Shukrullaev on 3/15/22.
//

import Foundation

public protocol OrdersRepository {
	func getOrdersList(completion: @escaping (Result<[Order], Error>) -> Void)
}

public class DefaultOrdersRepository: OrdersRepository {
	
	public init() {}

	public func getOrdersList(completion: @escaping (Result<[Order], Error>) -> Void) {
		let orders: [Order] = [
			Order(
				object: "",
				id: 1,
				table: "2",
				guests: 3,
				date: "23",
				items: [
					Item(object: "", id: 1, name: "122", price: 21, currency: "32", color: "red")
				]
			),
			Order(
				object: "",
				id: 1,
				table: "2",
				guests: 3,
				date: "23",
				items: [
					Item(object: "", id: 1, name: "122", price: 21, currency: "32", color: "red")
				]
			),
			Order(
				object: "",
				id: 1,
				table: "2",
				guests: 3,
				date: "23",
				items: [
					Item(object: "", id: 1, name: "122", price: 21, currency: "32", color: "red")
				]
			),
		]
		
		completion(.success(orders))
	}
}
